### PR TITLE
Commit Action Button - Fix separator when button is disabled

### LINKED ISSUE #154777: Commit Action Button - separator rendering while button is disabled
- Separator background is incorrect
- Separator cursor is incorrect

image

### PR DESCRIPTION
**Pull request by lszomoru**
Fix #154777

### CODE REVIEW

File: src/vs/base/browser/ui/button/button.css [hunk 1]
```diff
@@ -46,6 +46,11 @@
 	padding: 4px 0;
 }
 
+.monaco-button-dropdown .monaco-button-dropdown-separator.disabled {
+	opacity: 0.4 !important;
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/button/button.css:50]: Any chance we can get this without `!important`?

File: src/vs/base/browser/ui/button/button.ts [hunk 2]
```diff
@@ -299,6 +299,8 @@ export class ButtonWithDropdown extends Disposable implements IButton {
 	set enabled(enabled: boolean) {
 		this.button.enabled = enabled;
 		this.dropdownButton.enabled = enabled;
+
+		this.separatorContainer.classList.toggle('disabled', !enabled);
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/button/button.ts:303]: Should this class be on `this.element` instead?

File: src/vs/base/browser/ui/button/button.css [hunk 3]
```diff
@@ -46,6 +46,11 @@
...
+	cursor: default;
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/button/button.css:51]: Should the cursor style apply to the entire `.monaco-button-dropdown`?

File: src/vs/base/browser/ui/button/button.css (hunk 1 above)
Comment (joaomoreno) [reply src/vs/base/browser/ui/button/button.css:50]: Same comment for all other `!important` uses in this file. The issue with `!important` clauses is that they are sticky. Things that want to build on top of them will require `!important` as well. And then, once everything is `!important`, we no longer have a way to overwrite things, ie there's no `!superimportant`.

File: src/vs/base/browser/ui/button/button.css (hunk 3 above)
Comment (lszomoru) [reply src/vs/base/browser/ui/button/button.css]: 👍🏻

File: src/vs/base/browser/ui/button/button.css (hunk 1 above)
Comment (lszomoru) [reply src/vs/base/browser/ui/button/button.css]: Removed the `!important`.

**Review by joaomoreno — APPROVED** [review]

**Pull request merged**